executor/runtime/docker/docker.go

```diff
@@ -1361,48 +1358,50 @@ func (r *DockerRuntime) statusMonitor(cancel context.CancelFunc, c runtimeTypes.
 			log.Fatal("Got error while listening for docker events, bailing: ", err)
 		case event := <-eventChan:
 			log.Info("Got docker event: ", event)
-			if handleEvent(c, event, statusMessageChan) {
+			if handleDockerEvent(c, event, statusMessageChan) {
 				log.Info("Terminating docker status monitor because terminal docker event received")
 				return
 			}
 		}
 	}
 }
 
-// startOtherUserContainers launches the other user containers, only looking
-// any container objects in the pod *after* the first one, converting the
-// v1.Container spec into something docker can understand, and then
-// running that container.
-func (r *DockerRuntime) startOtherUserContainers(ctx context.Context, pod *v1.Pod, mainContainerID string) error {
-	if pod == nil || pod.Spec.Containers == nil {
-		return nil
-	}
-	otherUserContainers := pod.Spec.Containers[1:]
-	if len(otherUserContainers) == 0 {
-		return nil
+// startUserContainers launches all user containers (including the main one)
+// for containers other than the main one, it must pull and create them
+func (r *DockerRuntime) startUserContainers(ctx context.Context, pod *v1.Pod, mainContainerID string, tiniConn *net.UnixConn) error {
+	var otherUserContainers []v1.Container
+	if pod != nil && pod.Spec.Containers != nil {
+		otherUserContainers = pod.Spec.Containers[1:]
+	} else {
+		otherUserContainers = []v1.Container{}
```

Comment:
This is super nit-picky, and not totally necessary, *but* I know you're a Python person, and a PEP 20 enthusiast. So, in the interests of "simple is better" and "flat is better than nested" - maybe initialize `otherUserContainers` to an empty array, and you can remove this else?